tensornetwork/backends/backend_test.py

```diff
@@ -3,6 +3,8 @@
 import sys
 import pytest
 import numpy as np
+import tensornetwork as tn
+from tensornetwork import connect, contract
```

Comment:
Perfer not to import `tensornetwork` directly in unit tests outside of `tests/` directory (and prefer to ONLY do `import tensornetwork as tn` inside of `tests/`)

Reply:
Can you clarify the reasons? I don't mind changing, but looks like there's a subtlety to learn here I'm not fully aware of.

Reply:
Not really a subtlety more of a structure thing I'm weakly trying to enforce. Tests next to files are supposed to be totally self contained, just testing that one module. Thus, anything you need in the test should be imported in manually.

However, tests in `tests/` should be public facing user test. Basically, test the behavior of the library as a whole so that specific use cases are less likely to break. Think of difference like integration vs unit testing.

Like I said it's a weak enforcement, and that rule is likely broken somewhere in the codebase.

Reply:
got it.

Reply:
Tried to fix it in most places